### DISCUSSION TITLE
Can`t install gatsby plugins

### DISCUSSION THREAD

**Discussion by vadiminsk** [gatsbyjs/gatsby#30084] [Help]
## Description

Ubuntu 20.04, NPM 7.6.0 NodeJS 10.19.0
Have an error when trying to install filesystem plugin

### Steps to reproduce

npm install gatsby-source-filesystem

### Expected result

Should install plugin

### Actual result

`npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: gatsby-starter-hello-world@0.1.0
npm ERR! Found: gatsby@2.32.9
npm ERR! node_modules/gatsby
npm ERR!   gatsby@"^2.26.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer gatsby@"^3.0.0-next.0" from gatsby-source-filesystem@3.0.0
npm ERR! node_modules/gatsby-source-filesystem
npm ERR!   gatsby-source-filesystem@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/user/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2021-03-05T18_22_25_296Z-debug.log`

### Environment

It`s just my local env

**Comment by LekoArts** [comment]
Thank you for opening this!

You're on a v2 version of `gatsby` but want to install a version of `gatsby-source-filesystem` that requires v3 of `gatsby`. npm 7 doesn't handle this gracefully -- that's why we encourage to use `--legacy-peer-deps` flag: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#update-gatsby-version

Also, it's best if you also upgrade the `gatsby` version to v3. Then all installs of the plugins will work :)

We're marking this issue as answered and closing it for now but please feel free to comment here if you would like to continue this discussion. We also recommend heading over to our [communities](https://www.gatsbyjs.org/contributing/community/#where-to-get-support) if you have questions that are not bug reports or feature requests. We hope we managed to help and thank you for using Gatsby!
